src/golang/cmd/server/handler/v2/integrations_workflows_get.go

```diff
@@ -0,0 +1,136 @@
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		Map of integration ID to list of workflow IDs that use that integration.
+
+type integrationsWorkflowsGetArgs struct {
+	*aq_context.AqContext
+}
+
+type IntegrationsWorkflowsGetHandler struct {
+	handler.GetHandler
+
+	Database        database.Database
+	IntegrationRepo repos.Integration
+	OperatorRepo    repos.Operator
+}
+
+func (*IntegrationsWorkflowsGetHandler) Name() string {
+	return "WorkflowsByIntegrationGet"
+}
+
+func (h *IntegrationsWorkflowsGetHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, err
+	}
+
+	return &integrationsWorkflowsGetArgs{
+		AqContext: aqContext,
+	}, http.StatusOK, nil
+}
+
+func (h *IntegrationsWorkflowsGetHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*integrationsWorkflowsGetArgs)
+
+	integrations, err := h.IntegrationRepo.GetByUser(
+		ctx,
+		args.OrgID,
+		args.ID,
+		h.Database,
+	)
+	if err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to list integrations.")
+	}
+
+	response := make(map[uuid.UUID][]uuid.UUID, len(integrations))
+	for _, integration := range integrations {
+		workflowAndOperatorsList, err := h.fetchWorkflowIDsForIntegration(ctx, args.OrgID, &integration)
+		if err != nil {
+			return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to fetch workflow and operators for integration.")
+		}
+		response[integration.ID] = workflowAndOperatorsList
+	}
+	return response, http.StatusOK, nil
+}
+
+// fetchWorkflowIDsForIntegration returns a list of workflow IDs that use the given integration.
+func (h *IntegrationsWorkflowsGetHandler) fetchWorkflowIDsForIntegration(
+	ctx context.Context,
+	orgID string,
+	integration *models.Integration,
+) ([]uuid.UUID, error) {
+	integrationID := integration.ID
+
+	// If the requested integration is a cloud integration, substitute the cloud integration ID
+	// with the ID of the dynamic k8s integration.
+	if integration.Service == shared.AWS {
+		k8sIntegration, err := h.IntegrationRepo.GetByNameAndUser(
+			ctx,
+			fmt.Sprintf("%s:%s", integration.Name, dynamic.K8sIntegrationNameSuffix),
+			uuid.Nil,
+			orgID,
+			h.Database,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		integrationID = k8sIntegration.ID
+	}
+
+	operators, err := operator.GetOperatorsOnIntegration(
+		ctx,
+		integrationID,
+		h.IntegrationRepo,
+		h.OperatorRepo,
+		h.Database,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to retrieve operators.")
+	}
+
+	// Now, using the operators using this integration, we can infer all the workflows
+	// that also use this integration.
+	workflowIDs := make([]uuid.UUID, 0, 1)
+
+	operatorIDs := make([]uuid.UUID, 0, len(operators))
+	operatorByIDs := make(map[uuid.UUID]models.Operator, len(operators))
```

Comment:
We can use `go/lib/functional` library to do map / slice manipulations. I believe what you need is `maps.FromValues()`. We should add new ones if there's not a suitable one